packages/node-http-handler/src/node-http-handler.ts

```diff
@@ -192,4 +192,18 @@ export class NodeHttpHandler implements HttpHandler {
       writeRequestBodyPromise = writeRequestBody(req, request, this.config.requestTimeout).catch(_reject);
     });
   }
+
+  updateHttpClientConfig(key: keyof NodeHttpHandlerOptions, value: NodeHttpHandlerOptions[typeof key]): void {
+    this.config = undefined;
+    this.configProvider = this.configProvider.then((resolveConfig) => {
+      return {
+        ...resolveConfig,
+        [key]: value,
+      };
```

Comment:
this can be the same as the fetch http one with no copy.

Reply:
I tried and it throws `Type 'number | boolean' is not assignable to type 'never'.   Type 'number' is not assignable to type 'never'` error

Reply:
what about with 

```ts
(config: NodeHttpHandlerOptions) => {
    config[key] = value;
    return config;
}
```

Reply:
same error